### ISSUE TITLE
The GPU Memory Usage keeps increasing during training.

### ISSUE THREAD

**Issue by 910399614** [open-mmlab/mmdetection3d#1907]
**Notice**

There are several common situations in the reimplementation issues as below

1. Reimplement a model in the model zoo using the provided configs

**Checklist**

1. I have searched related issues but cannot get the expected help.
2. The issue has not been fixed in the latest version.

**Describe the issue**

The GPU Memory Usage keeps increasing during training of the PointRCNN model.

**Reproduction**

1. What command or script did you run?

```
python tools/train.py configs/point_rcnn/point_rcnn_2x8_kitti-3d-3classes.py
```

2. What config dir you run?

```
configs/point_rcnn/point_rcnn_2x8_kitti-3d-3classes.py
```

3. Did you make any modifications on the code or config? Did you understand what you have modified?
I change the samples_per_gpu from 2 to 1, and I understand it will reduce the GPU Memory Usage.

4. What dataset did you use?
KITTI

**Environment**

1. Please run `python mmdet3d/utils/collect_env.py` to collect necessary environment information and paste it here.

sys.platform: linux
Python: 3.7.0 (default, Oct  9 2018, 10:31:47) [GCC 7.3.0]
CUDA available: True
GPU 0,1,2,3: GeForce GTX 1080 Ti
CUDA_HOME: /usr/local/cuda
NVCC: Cuda compilation tools, release 10.2, V10.2.8
GCC: gcc (Ubuntu 6.5.0-2ubuntu1~18.04) 6.5.0 20181026
PyTorch: 1.10.1
PyTorch compiling details: PyTorch built with:
  - GCC 7.3
  - C++ Version: 201402
  - Intel(R) oneAPI Math Kernel Library Version 2021.4-Product Build 20210904 for Intel(R) 64 architecture applications
  - Intel(R) MKL-DNN v2.2.3 (Git Hash 7336ca9f055cf1bfa13efb658fe15dc9b41f0740)
  - OpenMP 201511 (a.k.a. OpenMP 4.5)
  - LAPACK is enabled (usually provided by MKL)
  - NNPACK is enabled
  - CPU capability usage: AVX2
  - CUDA Runtime 10.2
  - NVCC architecture flags: -gencode;arch=compute_37,code=sm_37;-gencode;arch=compute_50,code=sm_50;-gencode;arch=compute_60,code=sm_60;-gencode;arch=compute_61,code=sm_61;-gencode;arch=compute_70,code=sm_70;-gencode;arch=compute_75,code=sm_75;-gencode;arch=compute_37,code=compute_37
  - CuDNN 7.6.5
  - Magma 2.5.2
  - Build settings: BLAS_INFO=mkl, BUILD_TYPE=Release, CUDA_VERSION=10.2, CUDNN_VERSION=7.6.5, CXX_COMPILER=/opt/rh/devtoolset-7/root/usr/bin/c++, CXX_FLAGS= -Wno-deprecated -fvisibility-inlines-hidden -DUSE_PTHREADPOOL -fopenmp -DNDEBUG -DUSE_KINETO -DUSE_FBGEMM -DUSE_QNNPACK -DUSE_PYTORCH_QNNPACK -DUSE_XNNPACK -DSYMBOLICATE_MOBILE_DEBUG_HANDLE -DEDGE_PROFILER_USE_KINETO -O2 -fPIC -Wno-narrowing -Wall -Wextra -Werror=return-type -Wno-missing-field-initializers -Wno-type-limits -Wno-array-bounds -Wno-unknown-pragmas -Wno-sign-compare -Wno-unused-parameter -Wno-unused-variable -Wno-unused-function -Wno-unused-result -Wno-unused-local-typedefs -Wno-strict-overflow -Wno-strict-aliasing -Wno-error=deprecated-declarations -Wno-stringop-overflow -Wno-psabi -Wno-error=pedantic -Wno-error=redundant-decls -Wno-error=old-style-cast -fdiagnostics-color=always -faligned-new -Wno-unused-but-set-variable -Wno-maybe-uninitialized -fno-math-errno -fno-trapping-math -Werror=format -Wno-stringop-overflow, LAPACK_INFO=mkl, PERF_WITH_AVX=1, PERF_WITH_AVX2=1, PERF_WITH_AVX512=1, TORCH_VERSION=1.10.1, USE_CUDA=ON, USE_CUDNN=ON, USE_EXCEPTION_PTR=1, USE_GFLAGS=OFF, USE_GLOG=OFF, USE_MKL=ON, USE_MKLDNN=ON, USE_MPI=OFF, USE_NCCL=ON, USE_NNPACK=ON, USE_OPENMP=ON, 

TorchVision: 0.11.2
OpenCV: 4.6.0
MMCV: 1.5.2
MMCV Compiler: GCC 6.5
MMCV CUDA Compiler: 10.2
MMDetection: 2.25.2
MMSegmentation: 0.29.0
MMDetection3D: 1.0.0rc4+b27185a
spconv2.0: False

2. You may add addition that may be helpful for locating the problem, such as
   - How you installed PyTorch \[e.g., pip, conda, source\]
   - Other environment variables that may be related (such as `$PATH`, `$LD_LIBRARY_PATH`, `$PYTHONPATH`, etc.)

The version of the code looks like this
![~GHPW_W DR52TYJML_`II 7](https://user-images.githubusercontent.com/60724761/195079143-66730744-2e87-4317-b9c8-bc6d330a83b6.png)

**Results**

If applicable, paste the related results here, e.g., what you expect and what you get.
![AM(GXAKQXL1MS%{13AVJM3P](https://user-images.githubusercontent.com/60724761/195079487-bdcb05a6-a763-4971-ac77-1287055c5647.png)

**Issue fix**

If you have already identified the reason, you can provide the information here. If you are willing to create a PR to fix it, please also leave a comment here and that would be much appreciated!

**Comment by JingweiZhang12** [comment]
Did this phenomenon occur if you don't use DDP training？ Suggest you can profile it in a single GPU. Also, we'll study it and keep an eye on your later feedback.

**Comment by 910399614** [comment]
Thanks your replay. The same problem occurs when I use a single GPU to train the model (PointRCNN).




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jingwei ***@***.***&gt;; 
发送时间: 2022年10月13日(星期四) 上午9:53
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [open-mmlab/mmdetection3d] The GPU Memory Usage keeps increasing during training. (Issue #1907)





 
Did this phenomenon occur if you don't use DDP training？ Suggest you can profile it in a single GPU. Also, we'll study it and keep an eye on your later feedback.

**Comment by Clichong** [comment]
I had the same problem whether it was single gpu or multi gpu during training of the PointRCNN model. The memory usage will continue to increase until it overflows.

**Comment by VVsssssk** [comment]
Hi, I have fixed it in PR#1928, and I will check the model training result after modification later.

**Comment by 910399614** [comment]
> Hi, I have fixed it in PR#1928, and I will check the model training result after modification later.
